In [1]:
#Import des principales librairies Python
import pandas as pd
import plotly.figure_factory as ff
import requests
import seaborn as sns
%matplotlib inline

## Mission 1 : Positionnement de la voiture électrique (VE) en France<a id="voiture">

### Nettoyage des données<a id="nettoyage">

__`Traitement des données sur les immatriculations des véhicules neufs et d'occasions 2010 à 2019`__

In [2]:
#Chargement du jeu de données "immatriculation-voitures-neuves-source-energie-localisation"
xls = pd.ExcelFile('p8_data/immatriculation-voitures-neuves-source-energie-localisation.xlsx')
df2010 = pd.read_excel(xls, '2010 ', skiprows=2)
df2011 = pd.read_excel(xls, '2011', skiprows=2)
df2012 = pd.read_excel(xls, '2012', skiprows=2)
df2013 = pd.read_excel(xls, '2013', skiprows=2)
df2014 = pd.read_excel(xls, '2014', skiprows=2)
df2015 = pd.read_excel(xls, '2015', skiprows=2)
df2016 = pd.read_excel(xls, '2016', skiprows=2)
df2017 = pd.read_excel(xls, '2017', skiprows=2)
df2018 = pd.read_excel(xls, '2018', skiprows=2)
df2019 = pd.read_excel(xls, '2019', skiprows=2)

In [3]:
#Visualisation d'un jeu de données
df2011.head()

Régions Départements Essence + Superéthanol Unnamed: 3 Unnamed: 4  \
0     NaN          NaN                < 6 CV    6 à 7 CV  8 CV et +   
1  Alsace     Bas-Rhin                   6204       2309       1012   
2     NaN    Haut-Rhin                   5477       1916       1039   
3  Alsace        Total                  11681       4225       2051   
4     NaN          NaN                    NaN        NaN        NaN   

  Electricité-Essence Unnamed: 6  Gazole Unnamed: 8 Essence-GPL Unnamed: 10  \
0              < 6 CV  6 CV et +  < 6 CV  6 CV et +      < 6 CV   6 CV et +   
1                 209         37   10071      14324         166           1   
2                 124         34    8583      11368          92           1   
3                 333         71   18654      25692         258           2   
4                 NaN        NaN     NaN        NaN         NaN         NaN   

   Electricité Gazole-Electricité Unnamed: 13  Autres et non dét.    Total  
0          NaN             < 6 CV   6 CV et +                 NaN      NaN  
1         41.0                NaN         NaN                 5.0  34379.0  
2         23.0                NaN         NaN                 8.0  28665.0  
3         64.0                NaN         NaN                13.0  63044.0  
4          NaN                NaN         NaN                 NaN      NaN

In [4]:
#Ajout d'une colonne 'Année' par dataframe
df2010['Année'] = 2010
df2011['Année'] = 2011
df2012['Année'] = 2012
df2013['Année'] = 2013
df2014['Année'] = 2014
df2015['Année'] = 2015
df2016['Année'] = 2016
df2017['Année'] = 2017
df2018['Année'] = 2018
df2019['Année'] = 2019

In [5]:
#Concaténation par méthode .concat() des précédents jeux de données
immat_motor_puiss = pd.concat([df2010, df2011, df2012, df2013, df2014, df2015, df2016, df2017, df2018, df2019], 
                               sort=False)

In [6]:
#Premier aperçu du jeu de données 
immat_motor_puiss

Régions Départements Essence + Superéthanol  \
0                           NaN          NaN                < 6 CV    
1                        Alsace     Bas-Rhin                   6729   
2                           NaN    Haut-Rhin                   5545   
3                        Alsace        Total                  12274   
4                           NaN          NaN                    NaN   
..                          ...          ...                    ...   
121  Provence-Alpes-Côte d'Azur        Total                  58699   
122                         NaN          NaN                    NaN   
123                       Total          NaN                 700515   
124                         NaN          NaN                    NaN   
125       Source : SDES-RSVERO           NaN                    NaN   

    Unnamed: 3 Unnamed: 4 Electricité-Essence Unnamed: 6  Gazole Unnamed: 8  \
0     6 à 7 CV  8 CV et +              < 6 CV  6 CV et +  < 6 CV  6 CV et +   
1         2002        949                 162         53   11513      12841   
2         1602        864                  75         36    9892      10243   
3         3604       1813                 237         89   21405      23084   
4          NaN        NaN                 NaN        NaN     NaN        NaN   
..         ...        ...                 ...        ...     ...        ...   
121      29944      14882                5497       3728   12843      34218   
122        NaN        NaN                 NaN        NaN     NaN        NaN   
123     392605     173904               69043      44303  189450     546708   
124        NaN        NaN                 NaN        NaN     NaN        NaN   
125        NaN        NaN                 NaN        NaN     NaN        NaN   

    Essence-GPL Unnamed: 10  Electricité  Autres et non dét.      Total  \
0        < 6 CV   6 CV et +          NaN                 NaN        NaN   
1          1041           2          1.0                20.0    35313.0   
2           687           4          NaN                 8.0    28956.0   
3          1728           6          1.0                28.0    64269.0   
4           NaN         NaN          NaN                 NaN        NaN   
..          ...         ...          ...                 ...        ...   
121          49          55       4860.0                23.0   165769.0   
122         NaN         NaN          NaN                 NaN        NaN   
123         985        1053      42262.0               363.0  2172967.0   
124         NaN         NaN          NaN                 NaN        NaN   
125         NaN         NaN          NaN                 NaN        NaN   

     Année Gazole-Electricité Unnamed: 13  
0     2010                NaN         NaN  
1     2010                NaN         NaN  
2     2010                NaN         NaN  
3     2010                NaN         NaN  
4     2010                NaN         NaN  
..     ...                ...         ...  
121   2019              971.0         NaN  
122   2019                NaN         NaN  
123   2019            11776.0         NaN  
124   2019                NaN         NaN  
125   2019                NaN         NaN  

[1363 rows x 17 columns]

In [7]:
#Suppression des lignes dont les valeurs sont manquantes sur l'objet Series 'Départements'
#Suppression de la variable 'Unnamed: 13' non représentative dans l'échantillon
immat_motor_puiss.dropna(subset=['Départements'], axis=0, inplace=True)
immat_motor_puiss.drop(columns=['Unnamed: 13'], inplace=True)

In [8]:
#Renommage des colonnes prenant en compte les puissances de la ligne index 0
immat_motor_puiss.rename(columns={
    'Essence + Superéthanol': 'Essence + Superéthanol < 6CV', 'Unnamed: 3': 'Essence + Superéthanol 6 à 7CV',
    'Unnamed: 4': 'Essence + Superéthanol 8CV et +', 'Electricité-Essence': 'Hybride essence < 6CV',
    'Unnamed: 6': 'Hybride essence 6CV et +', 'Gazole': 'Gazole < 6CV', 'Unnamed: 8': 'Gazole 6CV et +',
    'Essence-GPL': 'Essence-GPL < 6CV', 'Unnamed: 10': 'Essence-GPL 6CV et +', 
    'Gazole-Electricité': 'Hybride gazole', 'Autres et non dét.': 'Autres'
                                  }, inplace=True)

In [9]:
#Réindexation du dataframe
immat_motor_puiss.reset_index(drop=True, inplace=True)

In [10]:
#Affichage des modalités liées aux départements
immat_motor_puiss['Départements'].unique()

array(['Bas-Rhin', 'Haut-Rhin', 'Total', 'Dordogne', 'Gironde', 'Landes',
       'Lot-et-Garonne', 'Pyrénées-Atlantiques', 'Allier', 'Cantal',
       'Haute-Loire', 'Puy-de-Dôme', 'Calvados', 'Manche', 'Orne',
       "Côte-d'Or", 'Nièvre', 'Saône-et-Loire', 'Yonne', "Côtes-d'Armor",
       'Finistère', 'Ille-et-Vilaine', 'Morbihan', 'Cher', 'Eure-et-Loir',
       'Indre', 'Indre-et-Loire', 'Loiret', 'Loir-et-Cher', 'Ardennes',
       'Aube', 'Haute-Marne', 'Marne', 'Corse-du-Sud', 'Haute-Corse',
       'Doubs', 'Haute-Saône', 'Jura', 'Territoire de Belfort', 'Eure',
       'Seine-Maritime', 'Essonne', 'Hauts-de-Seine', 'Paris',
       'Seine-et-Marne', 'Seine-Saint-Denis', 'Val-de-Marne',
       "Val-d'Oise", 'Yvelines', 'Aude', 'Gard', 'Hérault', 'Lozère',
       'Pyrénées-Orientales', 'Corrèze', 'Creuse', 'Haute-Vienne',
       'Meurthe-et-Moselle', 'Meuse', 'Moselle', 'Vosges', 'Ariège',
       'Aveyron', 'Gers', 'Haute-Garonne', 'Hautes-Pyrénées', 'Lot',
       'Tarn', 'Tarn-et-Gar

Il est nécessaire de supprimer la modalité 'Total' de la variable 'Départements'.

In [11]:
#Boucle itérative de suppression de la modalité 'Total'
for i, row in immat_motor_puiss.iterrows():
    if row['Départements'] == 'Total':
        immat_motor_puiss.drop(i, inplace=True)
        

#Autre méthode
'''immat_motor_puiss.drop(immat_motor_puiss[immat_motor_puiss['Départements'] == 'Total']
                        .index, axis=0, inplace = True) 
'''

"immat_motor_puiss.drop(immat_motor_puiss[immat_motor_puiss['Départements'] == 'Total']\n                        .index, axis=0, inplace = True) \n"

In [12]:
#Affichage des données
immat_motor_puiss

Régions      Départements Essence + Superéthanol < 6CV  \
0        Alsace          Bas-Rhin                         6729   
1           NaN         Haut-Rhin                         5545   
3     Aquitaine          Dordogne                         1909   
4           NaN           Gironde                         9690   
5           NaN            Landes                         2374   
...         ...               ...                          ...   
1141        NaN   Alpes-Maritimes                        13533   
1142        NaN  Bouches-du-Rhône                        20743   
1143        NaN      Hautes-Alpes                         1282   
1144        NaN               Var                        14884   
1145        NaN          Vaucluse                         6497   

     Essence + Superéthanol 6 à 7CV Essence + Superéthanol 8CV et +  \
0                              2002                             949   
1                              1602                             864   
3                               564                             172   
4                              1947                             864   
5                               528                             174   
...                             ...                             ...   
1141                           7208                            4326   
1142                          10641                            5058   
1143                            685                             292   
1144                           7364                            3396   
1145                           3330                            1481   

     Hybride essence < 6CV Hybride essence 6CV et + Gazole < 6CV  \
0                      162                       53        11513   
1                       75                       36         9892   
3                       48                       14         4275   
4                      134                       56        15277   
5                       38                        9         5191   
...                    ...                      ...          ...   
1141                  1302                      930         2145   
1142                  1988                     1289         5431   
1143                   159                      108          331   
1144                  1257                      913         2760   
1145                   585                      401         1535   

     Gazole 6CV et + Essence-GPL < 6CV Essence-GPL 6CV et +  Electricité  \
0              12841              1041                    2          1.0   
1              10243               687                    4          NaN   
3               3987               515                    6          NaN   
4              15267              2222                    3          NaN   
5               4628               420                    3          NaN   
...              ...               ...                  ...          ...   
1141            6847                12                   12       1231.0   
1142           13731                14                   15       2313.0   
1143            1044                 3                    4         85.0   
1144            7242                14                   10        716.0   
1145            4284                 6                    6        388.0   

      Autres    Total  Année Hybride gazole  
0       20.0  35313.0   2010            NaN  
1        8.0  28956.0   2010            NaN  
3        NaN  11490.0   2010            NaN  
4       12.0  45472.0   2010            NaN  
5        2.0  13367.0   2010            NaN  
...      ...      ...    ...            ...  
1141     2.0  37773.0   2019          225.0  
1142    20.0  61626.0   2019          383.0  
1143     NaN   4019.0   2019           26.0  
1144     NaN  38773.0   2019          217.0  
1145     1.0  18615.0   2019          101.0  

[960 rows x 16 columns]

In [13]:
#Affichage des modalités liées aux régions
immat_motor_puiss['Régions'].unique()

array(['Alsace', nan, 'Aquitaine', 'Auvergne', 'Basse-Normandie',
       'Bourgogne', 'Bretagne', 'Centre', 'Champagne-Ardenne', 'Corse',
       'Franche-Comté', 'Haute-Normandie', 'Île-de-France',
       'Languedoc-Roussillon', 'Limousin', 'Lorraine', 'Midi-Pyrénées',
       'Nord-Pas-de-Calais', 'Pays de la Loire', 'Picardie',
       'Poitou-Charentes', "Provence-Alpes-Côte d'Azur", 'Rhône-Alpes',
       'Centre-Val de Loire', 'Auvergne-Rhône-Alpes',
       'Bourgogne-Franche-Comté', 'Grand Est', 'Hauts-de-France',
       'Normandie', 'Nouvelle-Aquitaine', 'Occitanie'], dtype=object)

Les régions sont plus difficiles à gérer, car les observations vont de 2010 à 2019, et la Loi relative à la délimitation des régions de 2014 nous amène à avoir des parts de région en doublons. Dans l'immédiat les observations sont laissées dans l'état, la précision des observations par département est largement privilégiée dans notre contexte.

In [14]:
#Analyse du type de chaque colonne
immat_motor_puiss.dtypes

Régions                             object
Départements                        object
Essence + Superéthanol < 6CV        object
Essence + Superéthanol 6 à 7CV      object
Essence + Superéthanol 8CV et +     object
Hybride essence < 6CV               object
Hybride essence 6CV et +            object
Gazole < 6CV                        object
Gazole 6CV et +                     object
Essence-GPL < 6CV                   object
Essence-GPL 6CV et +                object
Electricité                        float64
Autres                             float64
Total                              float64
Année                                int64
Hybride gazole                      object
dtype: object

Notons que la plupart des données ne sont pas de type 'float' ou 'int', il est important de modifier ces types de données pour la suite des calculs.

In [15]:
#Changement du type des Series 'object' en 'float64' afin de faciliter les calculs
immat_motor_puiss[['Essence + Superéthanol < 6CV', 'Essence + Superéthanol 6 à 7CV', 
                    'Essence + Superéthanol 8CV et +', 'Hybride essence < 6CV', 'Hybride essence 6CV et +', 
                    'Gazole < 6CV', 'Gazole 6CV et +', 'Essence-GPL < 6CV','Essence-GPL 6CV et +',
                    'Hybride gazole']] = immat_motor_puiss[['Essence + Superéthanol < 6CV', 
                    'Essence + Superéthanol 6 à 7CV', 'Essence + Superéthanol 8CV et +', 'Hybride essence < 6CV',
                    'Hybride essence 6CV et +', 'Gazole < 6CV', 'Gazole 6CV et +', 'Essence-GPL < 6CV',
                    'Essence-GPL 6CV et +', 'Hybride gazole']].astype(float)

In [16]:
#Visualisation rapide du jeu de données
immat_motor_puiss.head()

Régions Départements  Essence + Superéthanol < 6CV  \
0     Alsace     Bas-Rhin                        6729.0   
1        NaN    Haut-Rhin                        5545.0   
3  Aquitaine     Dordogne                        1909.0   
4        NaN      Gironde                        9690.0   
5        NaN       Landes                        2374.0   

   Essence + Superéthanol 6 à 7CV  Essence + Superéthanol 8CV et +  \
0                          2002.0                            949.0   
1                          1602.0                            864.0   
3                           564.0                            172.0   
4                          1947.0                            864.0   
5                           528.0                            174.0   

   Hybride essence < 6CV  Hybride essence 6CV et +  Gazole < 6CV  \
0                  162.0                      53.0       11513.0   
1                   75.0                      36.0        9892.0   
3                   48.0                      14.0        4275.0   
4                  134.0                      56.0       15277.0   
5                   38.0                       9.0        5191.0   

   Gazole 6CV et +  Essence-GPL < 6CV  Essence-GPL 6CV et +  Electricité  \
0          12841.0             1041.0                   2.0          1.0   
1          10243.0              687.0                   4.0          NaN   
3           3987.0              515.0                   6.0          NaN   
4          15267.0             2222.0                   3.0          NaN   
5           4628.0              420.0                   3.0          NaN   

   Autres    Total  Année  Hybride gazole  
0    20.0  35313.0   2010             NaN  
1     8.0  28956.0   2010             NaN  
3     NaN  11490.0   2010             NaN  
4    12.0  45472.0   2010             NaN  
5     2.0  13367.0   2010             NaN

In [17]:
#Identification des valeurs manquantes
immat_motor_puiss.isna().sum()

Régions                            776
Départements                         0
Essence + Superéthanol < 6CV         0
Essence + Superéthanol 6 à 7CV       0
Essence + Superéthanol 8CV et +      0
Hybride essence < 6CV                0
Hybride essence 6CV et +             7
Gazole < 6CV                         0
Gazole 6CV et +                      0
Essence-GPL < 6CV                   71
Essence-GPL 6CV et +               113
Electricité                         81
Autres                             490
Total                                0
Année                                0
Hybride gazole                     192
dtype: int64

Rappelons le contexte, le dataframe est constitué de 8 jeux de données traitant des immatriculations des véhicules neufs et d'occassion en France Métropolitaine + Corse de 2010 à 2019. Il peut-être nécessaire d'avoir une vision plus précise sur le positionnement dans le temps et par catégorie de véhicules des observations manquantes.

In [18]:
#Suppression de la variable 'Autres' qui n'est pas exploitable dans l'état et non représentative 
immat_motor_puiss.drop(columns=['Autres'], inplace=True)

In [19]:
#Visualisation par restriction des valeurs manquantes de la variable 'Hybride essence 6CV et +'
immat_motor_puiss[(immat_motor_puiss['Hybride essence 6CV et +'].isna())]

Régions             Départements  \
63                          NaN                   Lozère   
102  Provence-Alpes-Côte d'Azur  Alpes-de-Haute-Provence   
182                         NaN                   Lozère   
186                         NaN                   Creuse   
201                         NaN          Tarn-et-Garonne   
221  Provence-Alpes-Côte d'Azur  Alpes-de-Haute-Provence   
538                         NaN                   Lozère   

     Essence + Superéthanol < 6CV  Essence + Superéthanol 6 à 7CV  \
63                          319.0                            61.0   
102                        1116.0                           193.0   
182                         282.0                            69.0   
186                         495.0                           108.0   
201                        1003.0                           203.0   
221                        1003.0                           232.0   
538                         301.0                            83.0   

     Essence + Superéthanol 8CV et +  Hybride essence < 6CV  \
63                              35.0                    3.0   
102                             89.0                   28.0   
182                             22.0                    8.0   
186                             44.0                   13.0   
201                             83.0                   31.0   
221                            112.0                   28.0   
538                             24.0                   17.0   

     Hybride essence 6CV et +  Gazole < 6CV  Gazole 6CV et +  \
63                        NaN         972.0            744.0   
102                       NaN        1969.0           1628.0   
182                       NaN         823.0            809.0   
186                       NaN        1140.0           1148.0   
201                       NaN        2289.0           2357.0   
221                       NaN        1780.0           1781.0   
538                       NaN         537.0            464.0   

     Essence-GPL < 6CV  Essence-GPL 6CV et +  Electricité   Total  Année  \
63                80.0                   NaN          NaN  2215.0   2010   
102              236.0                   3.0          NaN  5262.0   2010   
182               13.0                   NaN          NaN  2026.0   2011   
186               22.0                   NaN          1.0  2971.0   2011   
201               46.0                   1.0          1.0  6014.0   2011   
221               41.0                   3.0          2.0  4982.0   2011   
538                3.0                   3.0          3.0  1443.0   2014   

     Hybride gazole  
63              NaN  
102             NaN  
182             NaN  
186             NaN  
201             NaN  
221             NaN  
538             8.0

Notons que suivant l'année tous les types de véhicules n'étaient pas ou peu réprensentés suivant la localité. A ce stade de l'étude, les valeurs manquantes peuvent-être laissées en l'état.

In [20]:
#Création d'une variable 'Essence'
immat_motor = immat_motor_puiss.copy().reset_index()
immat_motor['Essence'] = immat_motor['Essence + Superéthanol < 6CV']\
                         + immat_motor['Essence + Superéthanol 6 à 7CV']\
                         + immat_motor['Essence + Superéthanol 8CV et +']

In [21]:
#Création d'une variable 'Gazole'
immat_motor['Gazole'] = immat_motor['Gazole < 6CV'] + immat_motor['Gazole 6CV et +']

In [22]:
#Création d'une variable 'Hybride'
immat_motor['Hybride'] = immat_motor['Hybride essence < 6CV']\
                       + immat_motor['Hybride essence 6CV et +']\
                       + immat_motor['Hybride gazole']

In [23]:
#Création d'une variable 'GPL'
immat_motor['GPL'] = immat_motor['Essence-GPL < 6CV'] + immat_motor['Essence-GPL 6CV et +']

In [24]:
#Projection sur les variables globales
immat_motor = immat_motor[['Régions', 'Départements', 'Année', 'Electricité', 'Essence', 'Gazole', 
                           'Hybride', 'GPL', 'Total']]

In [25]:
#Aperçu du jeu de données
immat_motor

Régions      Départements  Année  Electricité  Essence   Gazole  \
0       Alsace          Bas-Rhin   2010          1.0   9680.0  24354.0   
1          NaN         Haut-Rhin   2010          NaN   8011.0  20135.0   
2    Aquitaine          Dordogne   2010          NaN   2645.0   8262.0   
3          NaN           Gironde   2010          NaN  12501.0  30544.0   
4          NaN            Landes   2010          NaN   3076.0   9819.0   
..         ...               ...    ...          ...      ...      ...   
955        NaN   Alpes-Maritimes   2019       1231.0  25067.0   8992.0   
956        NaN  Bouches-du-Rhône   2019       2313.0  36442.0  19162.0   
957        NaN      Hautes-Alpes   2019         85.0   2259.0   1375.0   
958        NaN               Var   2019        716.0  25644.0  10002.0   
959        NaN          Vaucluse   2019        388.0  11308.0   5819.0   

     Hybride     GPL    Total  
0        NaN  1043.0  35313.0  
1        NaN   691.0  28956.0  
2        NaN   521.0  11490.0  
3        NaN  2225.0  45472.0  
4        NaN   423.0  13367.0  
..       ...     ...      ...  
955   2457.0    24.0  37773.0  
956   3660.0    29.0  61626.0  
957    293.0     7.0   4019.0  
958   2387.0    24.0  38773.0  
959   1087.0    12.0  18615.0  

[960 rows x 9 columns]

In [26]:
#Situation des valeurs manquantes par variable
immat_motor.isna().sum()

Régions         776
Départements      0
Année             0
Electricité      81
Essence           0
Gazole            0
Hybride         193
GPL             159
Total             0
dtype: int64

Rappelons que ces 960 observations s'échelonnent sur un timeframe de 2010 à 2019, il est normal d'avoir des NaN suivant le couple motorisation/année, également pouvant s'expliquer par des anomalies de recensement. Rappelons également que l'état des modalités de la variable 'Régions' est le résultat non seulement du remaniement des régions de 2014, mais également de la structuration des données réparties par départements. Dans l'immédiat seule l'approche départementale est pertinente, les modalités liées aux régions peuvent rester sans modification particulière.

In [27]:
#Affichage des observations par agrégation par Nombre d'immatriculation (sans cumul) selon les années
immat_annee = immat_motor.groupby('Année').sum().astype(int)
immat_annee

/tmp/ipykernel_17203/3094861806.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  immat_annee = immat_motor.groupby('Année').sum().astype(int)


Electricité  Essence   Gazole  Hybride    GPL    Total
Année                                                        
2010           187   568175  1556353        0  57892  2210186
2011          2630   575081  1557403        0  10381  2160928
2012          5661   476266  1345064    27817   1863  1857013
2013          8781   530292  1168193    46730   2753  1756952
2014         10567   589776  1119954    43153   2229  1765855
2015         17269   736339  1070443    60497   1516  1886229
2016         21758   875900  1027912    58302    300  1984471
2017         24910   993666   978184    81539    901  2079515
2018         30945  1174623   824472   106221   2150  2138600
2019         42262  1267024   736158   125122   2015  2172967

In [28]:
immat_annee.reset_index(inplace=True)
immat_annee['Année'] = pd.to_datetime(immat_annee['Année'], format='%Y')
immat_annee['Année'] = pd.date_range(start='31/12/2010', end='31/12/2019', freq ='Y')

/tmp/ipykernel_17203/3125475104.py:3: UserWarning: Parsing dates in DD/MM/YYYY format when dayfirst=False (the default) was specified. This may lead to inconsistently parsed dates! Specify a format to ensure consistent parsing.
  immat_annee['Année'] = pd.date_range(start='31/12/2010', end='31/12/2019', freq ='Y')


In [29]:
#Données des immatriculations par année sans cumul
immat_annee

Année  Electricité  Essence   Gazole  Hybride    GPL    Total
0 2010-12-31          187   568175  1556353        0  57892  2210186
1 2011-12-31         2630   575081  1557403        0  10381  2160928
2 2012-12-31         5661   476266  1345064    27817   1863  1857013
3 2013-12-31         8781   530292  1168193    46730   2753  1756952
4 2014-12-31        10567   589776  1119954    43153   2229  1765855
5 2015-12-31        17269   736339  1070443    60497   1516  1886229
6 2016-12-31        21758   875900  1027912    58302    300  1984471
7 2017-12-31        24910   993666   978184    81539    901  2079515
8 2018-12-31        30945  1174623   824472   106221   2150  2138600
9 2019-12-31        42262  1267024   736158   125122   2015  2172967

Voyons également en cumulé ce que représentent ces immatriculations…

In [30]:
#Sauvegarde 
immat_annee.to_csv('p8_datatable/immat_annee.csv')

In [31]:
#Affichage des observations par agrégation par somme cumulative selon les années
immat_motor_annee = immat_motor.groupby('Année').sum()
immat_motor_annee = immat_motor_annee.cumsum(axis=0).astype(int)
immat_motor_annee

/tmp/ipykernel_17203/2608568912.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  immat_motor_annee = immat_motor.groupby('Année').sum()


Electricité  Essence    Gazole  Hybride    GPL     Total
Année                                                          
2010           187   568175   1556353        0  57892   2210186
2011          2817  1143256   3113756        0  68273   4371114
2012          8478  1619522   4458820    27817  70136   6228127
2013         17259  2149814   5627013    74547  72889   7985079
2014         27826  2739590   6746967   117700  75118   9750934
2015         45095  3475929   7817410   178197  76634  11637163
2016         66853  4351829   8845322   236499  76934  13621634
2017         91763  5345495   9823506   318038  77835  15701149
2018        122708  6520118  10647978   424259  79985  17839749
2019        164970  7787142  11384136   549381  82000  20012716

In [32]:
immat_motor_annee.reset_index(inplace=True)
immat_motor_annee['Année'] = pd.to_datetime(immat_motor_annee['Année'], format='%Y')
immat_motor_annee['Année'] = pd.date_range(start='31/12/2010', end='31/12/2019', freq ='Y')

/tmp/ipykernel_17203/3138830374.py:3: UserWarning: Parsing dates in DD/MM/YYYY format when dayfirst=False (the default) was specified. This may lead to inconsistently parsed dates! Specify a format to ensure consistent parsing.
  immat_motor_annee['Année'] = pd.date_range(start='31/12/2010', end='31/12/2019', freq ='Y')


In [33]:
immat_motor_annee

Année  Electricité  Essence    Gazole  Hybride    GPL     Total
0 2010-12-31          187   568175   1556353        0  57892   2210186
1 2011-12-31         2817  1143256   3113756        0  68273   4371114
2 2012-12-31         8478  1619522   4458820    27817  70136   6228127
3 2013-12-31        17259  2149814   5627013    74547  72889   7985079
4 2014-12-31        27826  2739590   6746967   117700  75118   9750934
5 2015-12-31        45095  3475929   7817410   178197  76634  11637163
6 2016-12-31        66853  4351829   8845322   236499  76934  13621634
7 2017-12-31        91763  5345495   9823506   318038  77835  15701149
8 2018-12-31       122708  6520118  10647978   424259  79985  17839749
9 2019-12-31       164970  7787142  11384136   549381  82000  20012716

Pour information, il y a 32,2 millions de voitures particulières en circulation en France (chiffre 2016 de l’INSEE). Retenons que dans le cadre de l'étude, en cumulé de 2010 à 2019, plus de 20 Millions de voitures ont été immatriculées en France Métropolitaire + Corse.

In [34]:
#Affichage des principales statistiques descriptives
immat_motor_annee.describe()

Electricité       Essence        Gazole        Hybride           GPL  \
count      10.000000  1.000000e+01  1.000000e+01      10.000000     10.000000   
mean    54795.600000  3.570087e+06  7.002126e+06  192643.800000  73769.600000   
std     56097.250351  2.396082e+06  3.295060e+06  188478.496684   6985.889384   
min       187.000000  5.681750e+05  1.556353e+06       0.000000  57892.000000   
25%     10673.250000  1.752095e+06  4.750868e+06   39499.500000  70824.250000   
50%     36460.500000  3.107760e+06  7.282188e+06  147948.500000  75876.000000   
75%     85535.500000  5.097078e+06  9.578960e+06  297653.250000  77609.750000   
max    164970.000000  7.787142e+06  1.138414e+07  549381.000000  82000.000000   

              Total  
count  1.000000e+01  
mean   1.093579e+07  
std    5.887589e+06  
min    2.210186e+06  
25%    6.667365e+06  
50%    1.069405e+07  
75%    1.518127e+07  
max    2.001272e+07

In [35]:
#Sauvegarde 
immat_motor_annee.to_csv('p8_datatable/immat_motor_annee.csv')

In [36]:
#Affichage des observations par agrégation selon les départements
immat_motor_departement = immat_motor.groupby(['Départements', 'Année']).sum().reset_index()
immat_motor_departement

/tmp/ipykernel_17203/902243198.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  immat_motor_departement = immat_motor.groupby(['Départements', 'Année']).sum().reset_index()


Départements  Année  Electricité  Essence   Gazole  Hybride   GPL    Total
0            Ain   2010          0.0   5192.0  12670.0      0.0   0.0  18564.0
1            Ain   2011         10.0   5590.0  13504.0      0.0  97.0  19334.0
2            Ain   2012         21.0   4927.0  11208.0    246.0  18.0  16424.0
3            Ain   2013         55.0   5465.0  10299.0    352.0  23.0  16196.0
4            Ain   2014         60.0   5921.0   9408.0    314.0  14.0  15717.0
..           ...    ...          ...      ...      ...      ...   ...      ...
955     Yvelines   2015        935.0  27402.0  37209.0   2961.0  40.0  68551.0
956     Yvelines   2016        984.0  34940.0  34680.0   2392.0  12.0  73012.0
957     Yvelines   2017       1261.0  35521.0  33086.0   2943.0  24.0  72837.0
958     Yvelines   2018       1261.0  38254.0  29884.0   3409.0  50.0  72862.0
959     Yvelines   2019       1944.0  43613.0  27063.0   3844.0  52.0  76537.0

[960 rows x 8 columns]

In [37]:
#Sauvegarde
immat_motor_departement.to_csv('p8_datatable/immat_motor_departement.csv')

In [38]:
#Top 10 des meilleurs départements en 2019 pour les immatriculations VE
immat_motor_departement[immat_motor_departement['Année'] == 2019]\
.sort_values(by='Electricité', ascending=False)[:10]

Départements  Année  Electricité  Essence   Gazole  Hybride    GPL  \
719             Paris   2019       2564.0  31577.0  15652.0   5875.0   22.0   
129  Bouches-du-Rhône   2019       2313.0  36442.0  19162.0   3660.0   29.0   
959          Yvelines   2019       1944.0  43613.0  27063.0   3844.0   52.0   
449    Hauts-de-Seine   2019       1901.0  40375.0  37417.0   7145.0   45.0   
769             Rhône   2019       1616.0  32559.0  26270.0   3595.0   90.0   
689              Nord   2019       1304.0  52977.0  28070.0   4817.0  227.0   
369     Haute-Garonne   2019       1246.0  25364.0  13872.0   2588.0   63.0   
39    Alpes-Maritimes   2019       1231.0  25067.0   8992.0   2457.0   24.0   
339           Gironde   2019       1050.0  28166.0  16629.0   3274.0   67.0   
399      Haute-Savoie   2019        909.0  19477.0  10211.0   2384.0   30.0   

       Total  
719  55721.0  
129  61626.0  
959  76537.0  
449  86897.0  
769  64149.0  
689  87405.0  
369  43137.0  
39   37773.0  
339  49196.0  
399  33020.0

In [39]:
#Top 10 des meilleurs départements en 2019 pour les immatriculations VH
immat_motor_departement[immat_motor_departement['Année'] == 2019]\
.sort_values(by='Hybride', ascending=False)[:10]

Départements  Année  Electricité  Essence   Gazole  Hybride    GPL  \
449    Hauts-de-Seine   2019       1901.0  40375.0  37417.0   7145.0   45.0   
719             Paris   2019       2564.0  31577.0  15652.0   5875.0   22.0   
699              Oise   2019        632.0  82056.0  39519.0   5763.0   20.0   
689              Nord   2019       1304.0  52977.0  28070.0   4817.0  227.0   
959          Yvelines   2019       1944.0  43613.0  27063.0   3844.0   52.0   
129  Bouches-du-Rhône   2019       2313.0  36442.0  19162.0   3660.0   29.0   
769             Rhône   2019       1616.0  32559.0  26270.0   3595.0   90.0   
809    Seine-Maritime   2019        774.0  55099.0  27347.0   3350.0   57.0   
339           Gironde   2019       1050.0  28166.0  16629.0   3274.0   67.0   
829    Seine-et-Marne   2019        859.0  24117.0  14414.0   3245.0   46.0   

        Total  
449   86897.0  
719   55721.0  
699  127990.0  
689   87405.0  
959   76537.0  
129   61626.0  
769   64149.0  
809   86632.0  
339   49196.0  
829   42742.0